### PR TITLE
Add mono/platformer as a Mono port of 2d/platformer

### PR DESCRIPTION
**Pull request by zorbathut**
I wanted to play around with platformers in Godot, in C#. I liked the platformer demo and wanted to try adapting it for C#. I figure now that I've done so, I may as well provide it as a model for other people.

This is about 98% a line-for-line port. The only functional changes I've made:

* Coin detects the player by checking `body is Player`, instead of attempting to compare resource files
* Bullet detects shootable things and calls the relevant function via an interface IShootable, instead of looking for a function dynamically

Signals were converted to C#-standard naming. No further attempts were made to C#-ify the codebase.

This was implemented with a rather arbitrary snapshot of the current master branch, sometime after 3.1a1 but before a2/rc1/whatever comes next. If any breaking changes occur after the snapshot, this demo will need changes.

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @neikeq

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ export_presets.cfg
 # System/tool-specific ignores
 .directory
 *~
+.vs/
```
Comment (aaronfranke) [inline .gitignore]: Please ensure there are newlines at the end of each file.

File: mono/platformer/Bullet.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using Godot;
+
+class Bullet : RigidBody2D
+{
+    void _OnBulletBodyEnter(PhysicsBody2D body)
```
Comment (aaronfranke) [inline mono/platformer/Bullet.cs]: As per the [style guide](https://docs.godotengine.org/en/latest/getting_started/scripting/c_sharp/c_sharp_style_guide.html), use explicit access modifiers.

File: mono/platformer/Enemy.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        detect_wall_right = GetNode<RayCast2D>("detect_wall_right");
+        sprite = GetNode<Sprite>("sprite");
+    }
+
+    public override void _PhysicsProcess(float delta)
+    {
+        var new_anim = "idle";
+
+	    if (state == State.Walking)
+        {
```
Comment (aaronfranke) [inline mono/platformer/Enemy.cs]: Mixed spaces and tabs. Convert all tabs to four spaces as per the [style guide](https://docs.godotengine.org/en/latest/getting_started/scripting/c_sharp/c_sharp_style_guide.html).

**Comment by aaronfranke** [comment]
Unfortunately, this PR is just too old, and has not been updated with the changes that were requested.  
I'll close this PR, but feel free to discuss things below, or make a new one.

**Pull request closed without merging**